qt/src/SourceManager.cc

```diff
@@ -156,6 +168,26 @@ bool SourceManager::querySourcePassword(const QString& filename, QByteArray& pas
 	return success;
 }
 
+bool SourceManager::checkTextLayer(const QString& filename) const {
+	if(filename.endsWith(".pdf", Qt::CaseInsensitive)) {
+		std::unique_ptr<Poppler::Document> document(Poppler::Document::load(filename));
+		bool withoutText = true;
+		if(document) {
+			const int pagesNbr = document->numPages();
+
+			for (int i = 0; i < pagesNbr; ++i) {
+				QString text = document->page(i)->text(QRectF());
+				if(!text.isEmpty()) {
+					withoutText = false;
```

Comment:
There is no need to continue searching subsequent pages once you hit a page with text. Just return false here, and you can save yourself some lines of code since you don't need "bool withoutText" at all. (Note: shorter code is always a plus!)

Reply:
Fixed.